File: spring-integration/message-transformers/basic-transformer/src/main/java/com/ddubson/integration/services/StringPrintService.java
package com.ddubson.integration.services;

import com.ddubson.LogAdapter;
import org.springframework.messaging.Message;
import org.springframework.stereotype.Component;

import static com.ddubson.ANSIColor.ANSI_GREEN;
import static java.lang.String.format;

@Component
public class StringPrintService implements PrintService {
    private LogAdapter logAdapter;

    public StringPrintService(LogAdapter logAdapter) {
        this.logAdapter = logAdapter;
    }

    @Override
    public void print(Message<String> msg) {
        logAdapter.info("[2] Transformed message (uppercase): " + format("[%s] %s", Thread.currentThread().getName(),
                msg.getPayload()), ANSI_GREEN);
    }
}

File: spring-integration/routing-messages/payload-type-router/src/main/kotlin/com/ddubson/integration/services/StringPrintService.kt
package com.ddubson.integration.services

import com.ddubson.ANSIColor.ANSI_GREEN
import com.ddubson.LogAdapter
import org.springframework.messaging.Message
import org.springframework.stereotype.Component

@Component("stringPrintService")
class StringPrintService(private val logAdapter: LogAdapter): PrintService {
    override fun print(msg: Message<*>) {
        logAdapter.info("[2] String service: [${Thread.currentThread().name}] ${msg.payload}", ANSI_GREEN)
    }
}